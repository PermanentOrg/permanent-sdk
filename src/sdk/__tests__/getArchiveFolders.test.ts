import nock from 'nock';
import { getArchiveFolders } from '..';

describe('getArchiveFolders', () => {
  it('should return an array of folders', async () => {
    nock('https://permanent.local')
      .get('/api/folder/getRoot')
      .query({
        archiveId: 1,
      })
      .replyWithFile(
        200,
        `${__dirname}/fixtures/getArchiveFolders/archiveRoot.json`,
        {
          'Content-Type': 'application/json',
        },
      );

    const archives = await getArchiveFolders(
      {
        bearerToken: '12345',
        baseUrl: 'https://permanent.local/api',
      },
      {
        archiveId: 1,
      },
    );

    expect(archives).toMatchSnapshot();
  });
});
