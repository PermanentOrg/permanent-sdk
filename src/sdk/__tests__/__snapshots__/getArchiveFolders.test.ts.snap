// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getArchiveFolders should return an array of folders 1`] = `
[
  {
    "archiveRecords": [],
    "createdAt": 2022-12-16T15:45:10.000Z,
    "displayDate": 2022-12-16T15:45:10.000Z,
    "fileSystemCompatibleName": "Apps",
    "folders": [],
    "id": 2,
    "name": "Apps",
    "size": 0,
    "updatedAt": 2022-12-16T15:45:11.000Z,
  },
  {
    "archiveRecords": [],
    "createdAt": 2022-12-16T15:45:10.000Z,
    "displayDate": 2022-12-16T15:45:10.000Z,
    "fileSystemCompatibleName": "My Files",
    "folders": [],
    "id": 3,
    "name": "My Files",
    "size": 0,
    "updatedAt": 2022-12-16T16:15:13.000Z,
  },
  {
    "archiveRecords": [],
    "createdAt": 2022-12-16T15:45:11.000Z,
    "displayDate": 2022-12-16T15:45:11.000Z,
    "fileSystemCompatibleName": "Public",
    "folders": [],
    "id": 4,
    "name": "Public",
    "size": 0,
    "updatedAt": 2022-12-16T15:45:13.000Z,
  },
]
`;
